specification/compute/resource-manager/Microsoft.Compute/stable/2022-03-02/DiskRP/diskRPCommon.json

```diff
@@ -0,0 +1,593 @@
+        },
+        "logicalSectorSize": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Logical sector size in bytes for Ultra disks. Supported values are 512 ad 4096. 4096 is the default."
+        },
+        "securityDataUri": {
+          "type": "string",
+          "description": "If createOption is ImportSecure, this is the URI of a blob to be imported into VM guest state."
+        }
+      },
+      "required": [
+        "createOption"
+      ],
+      "description": "Data used when creating a disk."
+    },
+    "ImageDiskReference": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "A relative uri containing either a Platform Image Repository, user image, or Azure Compute Gallery image reference."
+        },
+        "sharedGalleryImageId": {
+          "type": "string",
+          "description": "A relative uri containing a direct shared Azure Compute Gallery image reference."
+        },
+        "communityGalleryImageId": {
+          "type": "string",
+          "description": "A relative uri containing a community Azure Compute Gallery image reference."
+        },
+        "lun": {
+          "type": "integer",
+          "format": "int32",
+          "description": "If the disk is created from an image's data disk, this is an index that indicates which of the data disks in the image to use. For OS disks, this field is null."
+        }
+      },
+      "description": "The source image used for creating the disk."
+    },
+    "PurchasePlan": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The plan ID."
+        },
+        "publisher": {
+          "type": "string",
+          "description": "The publisher ID."
+        },
+        "product": {
+          "type": "string",
+          "description": "Specifies the product of the image from the marketplace. This is the same value as Offer under the imageReference element."
+        },
+        "promotionCode": {
+          "type": "string",
+          "description": "The Offer Promotion Code."
+        }
+      },
+      "required": [
+        "publisher",
+        "name",
+        "product"
+      ],
+      "description": "Used for establishing the purchase context of any 3rd Party artifact through MarketPlace."
+    },
+    "CopyCompletionError": {
+      "type": "object",
+      "properties": {
+        "errorCode": {
+          "type": "string",
+          "description": "Indicates the error code if the background copy of a resource created via the CopyStart operation fails.",
+          "enum": [
+            "CopySourceNotFound"
+          ],
+          "x-ms-enum": {
+            "name": "CopyCompletionError",
```

Comment:
We have a model and an enum type with the same name, this is not allowed. Please either rename the model on line 418 or rename the type name of the enumeration on line 428

Reply:
Created a PR - https://github.com/Azure/azure-rest-api-specs/pull/19145

Reply:
New PR - https://github.com/Azure/azure-rest-api-specs/pull/19160